src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/ContactAddressService.kt

```
@@ -108,6 +110,69 @@ class ContactAddressService(
     return UpdateAddressResponse(contactAddressRepository.saveAndFlush(changedContactAddress).toModel(), updatedAddressIds)
   }
 
+  fun patch(contactId: Long, contactAddressId: Long, request: PatchContactAddressRequest): UpdateAddressResponse {
+    val contact = validateContactExists(contactId)
+    val existing = validateExistingAddress(contactAddressId)
+    request.cityCode.ifPresent { validateCityCode(it) }
+    request.countyCode.ifPresent { validateCountyCode(it) }
+    request.countryCode.ifPresent { validateCountryCode(it) }
+    val updatedAddressIds = mutableSetOf<Long>()
+
+    request.primaryAddress.ifPresent {
+      if (it && !existing.primaryAddress) {
+        updatedAddressIds += contactAddressRepository.resetPrimaryAddressFlagForContact(contactId)
+      }
+    }
+
+    request.mailFlag.ifPresent {
+      if (it && !existing.mailFlag) {
+        updatedAddressIds += contactAddressRepository.resetMailAddressFlagForContact(contactId)
+      }
+    }
+
+    if (contact.contactId != existing.contactId) {
+      logger.error("Contact address update specified for a contact not linked to this address")
```

This probably wants to be placed before doing any of the updates for primary/mail addresses, and probably before the city/county/country code validation?

---

I have updated this now

---

Thanks. Just means it fails faster, no need to do the other stuff if this check fails.